### DISCUSSION TITLE
How can I integrate filament with existing APIs

### DISCUSSION THREAD

**Discussion by agonkovaci** [filamentphp/filament#12777] [Help]
### Package

Panel builder

### Package Version

3.2.75

### How can we help you?

I have existing API app that I use for mobile applications. I want to integrate filament for some pretty views, but I want to integrate filament (create,update,list) with my existing API and not to default filament CRUD operation directly with eloquent models. For example I want to send submit form to my API p.s

`Route::post('/teachers', [TeacherController::class, 'store']);`

and edit to

`Route::put('/teachers/{id}', [TeacherController::class, 'update']);`

is it possible ? if Yes how can I do it, because Im not familiar with filament.

**Comment by leandrocfe** [comment]
You can use sushi https://filamentphp.com/community/how-to-consume-an-external-api-with-filament-tables
and customize the create, update pages https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-the-creation-process

**Reply by agonkovaci** [reply]
Alright, this function accepts array instead of request and return the model instance. I made the modifications to be fit with my APIs. But I get some other error.

`Missing required parameter for [Route: filament.superAdmin.resources.users.edit] [URI: superAdmin/users/{record}/edit] [Missing parameter: record].`

The `handleRecordCreation` method looks like

```
protected function handleRecordCreation(array $data): \Illuminate\Database\Eloquent\Model
    {
        $user = new UserService();
        $request = new Request($data);
        $response = $user->insertUserByRole($request,$data['role']);
        return new User((array)$response->getData()->user);
    }
```

 Should I make any changes on other place ?

Thank you for your reply

**Reply by leandrocfe** [reply]
Edit/View page expects `$record` (/1/edit). You can customize the mount method
```php
public function mount(int | string $record): void
{
    //...
}
```

**Comment by iafilin** [comment]
https://github.com/iafilin/eloquenthttpadapter
